main.go

```diff
@@ -70,6 +70,7 @@ func main() {
 }
 
 const DEFAULT_CHECK_TIMEOUT = 60
+const DEFAULT_IPNI_INDEXER = "https://cid.contact"
```

Comment:
@lidel mentioned we should use `https://cid.contact/routing/v1/providers/` as our default IPNI_INDEXER url

Reply:
Yeah, IMO we should be using routing-v1 APIs rather than the IPNI-specific ones mentioned here unless there's a particularly interesting debugging reason not to. This is primarily because codebases like kubo, rainbow, and helia use routing-v1 to interact with IPNI and so if there's an issue with say cid.contact's IPNI API that's likely more important to know about than if the IPNI-specific `/cid` or `/mh` APIs are working.

@gammazero IIUC you were pointing @SgtPooki at this code so wanted to check with you in case I'm missing something.

Note: If in the future IPNI evolves into a more federated form where there's some logic involved in being able to contact multiple instances that would certainly warrant more custom code here, but my understanding is that's not where we are today.

For now, maybe leveraging the code from boxo like the links below might be better:

https://github.com/ipfs/boxo/blob/171b0b72118578e2f42f7e5aa45b52ad33a04231/routing/http/client/client.go#L183
https://github.com/ipfs/boxo/blob/171b0b72118578e2f42f7e5aa45b52ad33a04231/routing/http/contentrouter/contentrouter.go#L180

I'd probably start with the first iteration just focusing on if there are Bitswap records (since that's the only data transfer protocol we're checking here anyway) as is the case in the code from the second link. As more of the widely used clients start adopting other transports (e.g. HTTP trustless gateway directly or over libp2p) we should add them here too.

To some extent what we're doing is trying to meet the expectations of users who are coming here likely after using some service like dweb.link or a tool like IPFS Desktop / kubo, so maybe having the UI give a little info on what's supported by the tool vs not would be helpful . Some text around this in the dweb.link context which is similar to here is at https://docs.ipfs.tech/concepts/public-utilities/#supported-features

Reply:
Handled.